src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -117,94 +125,148 @@ public String getScript() {
             }
             scriptEncodingCharset = zOSSystemEncodingCharset.name();
         }
-        
+
         ShellController c = new ShellController(ws,(os == OsType.ZOS));
         FilePath shf = c.getScriptFile(ws);
 
         shf.write(script, scriptEncodingCharset);
 
         final Jenkins jenkins = Jenkins.getInstance();
-        String interpreter = "";
+        String shell = null;
         if (!script.startsWith("#!")) {
-            String shell = jenkins.getDescriptorByType(Shell.DescriptorImpl.class).getShell();
+            shell = jenkins.getDescriptorByType(Shell.DescriptorImpl.class).getShell();
             if (shell == null) {
                 // Do not use getShellOrDefault, as that assumes that the filesystem layout of the agent matches that seen from a possibly decorated launcher.
                 shell = "sh";
             }
-            interpreter = "'" + shell + "' -xe ";
         } else {
             shf.chmod(0755);
         }
 
         String scriptPath = shf.getRemote();
-        List<String> args = new ArrayList<>();
 
-        if (os != OsType.DARWIN) { // JENKINS-25848
-            args.add("nohup");
+        // The temporary variable is to ensure JENKINS_SERVER_COOKIE=durable-… does not appear even in argv[], lest it be confused with the environment.
+        envVars.put(cookieVariable, "please-do-not-kill-me");
+
+        String arch = ws.act(new getArchitecture());
```

Comment:
Would be good to have a single `OsTypeAndArch getOsAndArchitecture` method so that we only have to make one remote call using `ws.act`, not two.

Reply:
Done in f2ac93b.